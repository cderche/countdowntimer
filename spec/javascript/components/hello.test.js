import React from 'react'
import Hello from '../../../app/javascript/components/Hello'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

test('display default name', () => {
  const { container } = render(<Hello />)
  expect(screen.getByRole('heading')).toHaveTextContent('Hello David!')
  expect(container).toMatchSnapshot()
})

test('display prop name', () => {
  const { container } = render(<Hello name='John' />)
  expect(screen.getByRole('heading')).toHaveTextContent('Hello John!')
  expect(container).toMatchSnapshot()
})