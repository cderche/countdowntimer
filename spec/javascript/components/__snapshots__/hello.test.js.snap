// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display default name 1`] = `
<div>
  <h1>
    Hello 
    David
    !
  </h1>
</div>
`;

exports[`display prop name 1`] = `
<div>
  <h1>
    Hello 
    John
    !
  </h1>
</div>
`;
